src/image_builder.py

```diff
@@ -154,109 +156,198 @@ def image_builder(buildspec):
+    #After the build, display logs/summary for all the images.
+    show_build_logs(ALL_IMAGES)
+    show_build_summary(ALL_IMAGES)
+    is_any_build_failed, is_any_build_failed_size_limit = show_build_errors(ALL_IMAGES)
+
+    #From all images, filter the images that were supposed to be built and upload their metrics
+    BUILT_IMAGES = [image for image in ALL_IMAGES if image.to_build]
+
+    FORMATTER.title("Upload Metrics")
+    # change logic here. upload metrics only for the Conclusion stage image
+    upload_metrics(BUILT_IMAGES, BUILDSPEC, is_any_build_failed, is_any_build_failed_size_limit)
+
+    FORMATTER.title("Setting Test Env")
+    # Set environment variables to be consumed by test jobs
+    test_trigger_job = utils.get_codebuild_project_name()
+    # Tests should only run on images that were pushed to the repository
+    utils.set_test_env(
+        IMAGES_TO_PUSH,
+        BUILD_CONTEXT=os.getenv("BUILD_CONTEXT"),
+        TEST_TRIGGER=test_trigger_job,
+    )
+
+def get_conclusion_stage_image_object(initial_stage_image_object):
+    # Check if this is only required for mainline
+    # if build_context == "MAINLINE":
+    conclusion_stage_image_object = None
+    conclusion_stage_image_object = ConclusionStageImage(
+        info=initial_stage_image_object.info,
+        dockerfile=os.path.join(os.sep, os.getenv("ROOT_FOLDER_PATH"), "src", "Dockerfile.multipart"),
+        repository=initial_stage_image_object.repository,
+        tag=initial_stage_image_object.tag,
+        to_build=initial_stage_image_object.to_build,
+        stage=constants.CONCLUSION_STAGE,
+    )
+    initial_stage_image_object.to_push = False
+    return conclusion_stage_image_object
+
+def show_build_logs(images):
+
+    FORMATTER.title("Build Logs")
+
+    if not os.path.isdir("logs"):
+        os.makedirs("logs")
+
+    for image in images:
+        image_description = f"{image.name}-{image.stage}"
+        FORMATTER.title(image_description)
+        FORMATTER.table(image.info.items())
+        FORMATTER.separator()
+        FORMATTER.print_lines(image.log)
+        with open(f"logs/{image_description}", "w") as fp:
+            fp.write("/n".join(image.log))
+            image.summary["log"] = f"logs/{image_description}"
 
-        FORMATTER.title("Build Logs")
+def show_build_summary(images):
 
-        if not os.path.isdir("logs"):
-            os.makedirs("logs")
+    FORMATTER.title("Summary")
 
-        for image in IMAGES:
-            FORMATTER.title(image.name)
-            FORMATTER.table(image.info.items())
-            FORMATTER.separator()
-            FORMATTER.print_lines(image.log)
-            with open(f"logs/{image.name}", "w") as fp:
-                fp.write("/n".join(image.log))
-                image.summary["log"] = f"logs/{image.name}"
+    for image in images:
+        FORMATTER.title(image.name)
+        FORMATTER.table(image.summary.items())
 
-        FORMATTER.title("Summary")
+def show_build_errors(images):
```

Comment:
Please add a docstring to this function.

Reply:
Done.